app/validation/validators.py

```diff
@@ -171,9 +173,12 @@ def __call__(self, form, field):
         if hasattr(form, 'day'):
```

Comment:
Consider a refactor to avoid numerous points at which the same exception is raised:

```
def __call__(self, form, field):
    invalid_year = not form.year.data
    invalid_month = hasattr(form, 'month') and not form.month.data
    invalid_day = hasattr(form, 'day') and not form.day.data

    if invalid_year or invalid_month or invalid_year:
        raise validators.StopValidation(self.message)
```

Reply:
@nealedj So this refactor wouldn't quite work. This check is to check you've entered at least one field in a mandatory date. Not if it's a valid date. The DateCheck below checks the format. I've come up with a cleaner way to do this though so will push that soon.